examples/basic/example_test.go

```diff
@@ -47,10 +47,16 @@ func ExampleBasicPlugin() {
 
 	for i := 0; i < numNodes; i++ {
 		builder := network.NewBuilder()
-		builder.SetKeys(ed25519.RandomKeyPair())
-		builder.SetAddress(network.FormatAddress("tcp", host, uint16(startPort+i)))
-
-		builder.AddPlugin(new(discovery.Plugin))
+		address := network.FormatAddress("tcp", host, uint16(startPort+i))
+		kp, id := discovery.GenerateKeyPairAndID(address)
+		builder.SetKeys(kp)
+		builder.SetAddress(address)
+		builder.SetNodeIDNonce(peer.GetNonce(id))
```

Comment:
Hmm, we shouldn't keep S/Kademlia-related methods in network.Builder; it's better we have discovery.GenerateKeypairAndID(address) already generate the nonce.

Reply:
the builder is the one that creates the Peer though so it needs to know at least whether the peer needs to have a s/kademlia compatible node id